packages/yew/src/functional/hooks/use_force_update.rs

```diff
@@ -23,12 +23,30 @@ impl UseForceUpdate {
     }
 }
 
-// #![feature(fn_traits)] // required nightly feature to make UseForceUpdate callable directly
-// impl Fn<()> for UseForceUpdate {
-//     extern "rust-call" fn call(&self, _args: ()) {
-//         self.force_update()
-//     }
-// }
+#[cfg(feature = "nightly")]
+mod feat_nightly {
+    use super::*;
+
+    impl FnOnce<()> for UseForceUpdate {
```

Comment:
Why not `type UseForceUpdate = Rc<dyn Fn()>`?

If we do not wish users to assume the underlying type of the handle, we can use `type UseForceUpdate = impl Fn();` for nightly.

Also maybe should be named `UseForceUpdateHandle` so it's consistent with other hooks.

Reply:
> Why not `type UseForceUpdate = Rc<dyn Fn()>`?

@WorldSEnder can better answer that. They added it in https://github.com/yewstack/yew/pull/2586
 
> If we do not wish users to assume the underlying type of the handle, we can use `type UseForceUpdate = impl Fn();` for nightly.

There's no point of doing that since the underlying type is `Rc<dyn Fn()>`. We can't use `impl Fn()` there because it is created as:

https://github.com/yewstack/yew/blob/015412e6ff7948c63b7ef343742075a247d1a2de/packages/yew/src/functional/mod.rs#L356

Replacing that with `impl Fn` fails because the type alias can't capture the type parameter `T`.

> Also maybe should be named `UseForceUpdateHandle` so it's consistent with other hooks.

Why not just expose `ReRender` type instead of having
```
type UseForceUpdateHandle = ReRender;
```

Reply:
The reason was to not expose the fact that it's an `Rc<_>` since that's an implementation detail and the exact smart pointer in use might change. I'm not beefed up on the ABI of `impl Trait` for this case, but I think it makes it impossible to add other methods (and at least `Sync`/`Send` would have to be added anyway). That part is easier to control with a struct.

> Also maybe should be named UseForceUpdateHandle

No objections. At the time I thought about `Handle` more for stateful things and just named it after the hook, but some consistency would be nice.

Reply:
I agree with having the option to add more methods on the type